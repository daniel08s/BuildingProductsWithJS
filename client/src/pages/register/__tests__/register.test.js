/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import RegisterWrapper, {Register} from '../index';

// create mock store
const mockStore = configureMockStore();

test('# Register page wrapper', () => {
  const store = mockStore({auth: {redirectToLogin: false}});

  const wrapper = shallow(<RegisterWrapper store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Register page', () => {
  const newUser = {
    login: 'testuser',
    password: '123',
    passwordRepeat: '123',
  };
  const navToLogin = () => expect(true).toBeTruthy();
  const onRegisterClick = ({login, password, passwordRepeat}) => {
    expect(login).toBe(newUser.login);
    expect(password).toBe(passwordRepeat);
    expect(password).toBe(newUser.password);
    expect(passwordRepeat).toBe(newUser.passwordRepeat);
  };

  const component = (
    <Register
      onRegisterClick={onRegisterClick}
      navToLogin={navToLogin}
      redirectToLogin={false}
    />
  );

  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  const app = mount(component);
  app.find('#inputUsername').getDOMNode().value = newUser.login;
  app.find('#inputPassword').getDOMNode().value = newUser.password;
  app.find('#inputPasswordRepeat').getDOMNode().value = newUser.passwordRepeat;

  // click button
  app.find('button').simulate('click');
});
