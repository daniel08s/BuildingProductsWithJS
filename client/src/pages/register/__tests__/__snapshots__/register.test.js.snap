// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Register page 1`] = `
<form
  className="form-signin"
>
  <h1
    className="h1 mb-3 font-weight-normal"
  >
    Experts portal
  </h1>
  <p
    className="h4 mb-3 font-weight-normal"
  >
    Please fill the form to register
  </p>
  <label
    className="sr-only"
    htmlFor="inputUsername"
  />
  <input
    className="form-control"
    id="inputUsername"
    placeholder="Username"
    required={true}
    type="text"
  />
  <label
    className="sr-only"
    htmlFor="inputPassword"
  />
  <input
    className="form-control"
    id="inputPassword"
    placeholder="Password"
    required={true}
    type="password"
  />
  <label
    className="sr-only"
    htmlFor="inputPasswordRepeat"
  />
  <input
    className="form-control"
    id="inputPasswordRepeat"
    placeholder="Password Confirmation"
    required={true}
    type="password"
  />
  <p />
  <button
    className="btn btn-lg btn-primary btn-block"
    onClick={[Function]}
    type="submit"
  >
    Register
  </button>
  <p
    className="h5 font-weight-normal"
  >
    Already registered? 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/login"
    >
      Login here.
    </Link>
  </p>
  <p
    className="mt-5 mb-3 text-muted"
  >
    Â© 2017-2018
  </p>
</form>
`;

exports[`# Register page wrapper 1`] = `
<Register
  navToLogin={[Function]}
  onRegisterClick={[Function]}
  redirectToLogin={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
